hack/release-manifest.py

```diff
@@ -54,51 +61,151 @@ def create_cluster_role_yaml(csv):
-}
-    return role_binding
+    return {
+        "apiVersion" : 'rbac.authorization.k8s.io/v1',
+        "kind" : 'ClusterRoleBinding',
+        "metadata" : {
+            "name" : 'service-binding-operator',
+        },
+        "roleRef" : {
+            "apiGroup" : 'rbac.authorization.k8s.io',
+            "kind" : 'ClusterRole',
+            "name" : 'service-binding-operator',
+        },
+        "subjects": [
+            {
+                "kind": 'ServiceAccount',
+                "name": 'service-binding-operator',
+                "namespace": 'service-binding-operator',
+            },
+        ],
+    }
 
 def create_operator_deployment(csv):
     depl = {
-    "apiVersion" : 'apps/v1',
-    "kind" : 'Deployment',
-    "metadata" : {
-        "name" : 'service-binding-operator',
-        "annotations": {"olm.targetNamespaces" : ""},
-        "namespace" : 'service-binding-operator',
-    },
-}
+        "apiVersion" : 'apps/v1',
+        "kind" : 'Deployment',
+        "metadata" : {
+            "name" : 'service-binding-operator',
+            "annotations": {"olm.targetNamespaces" : ""},
+            "namespace" : 'service-binding-operator',
+        },
+    }
+
     with open(csv, 'r') as f:
         data = yaml.load(f)
     roles = data["spec"]["install"]["spec"]["deployments"][0]
     del roles['name']
     depl.update(roles)
     return depl
 
+
+def create_cluster_issuer():
+    return {
+        "apiVersion": "cert-manager.io/v1",
+        "kind": "ClusterIssuer",
+        "metadata": {
+            "name": "selfsigned-issuer",
+        },
+        "spec": {
+            "selfSigned": {}
+        }
+    }
+
+def create_webhook_certificate():
+    return {
+        "apiVersion": "cert-manager.io/v1",
+        "kind": "Certificate",
+        "metadata": {
+            "name": "service-binding-operator-serving-cert",
+            "namespace": "service-binding-operator",
+        },
+        "spec": {
+            "dnsNames": [
+                "service-binding-operator-webhook-service.service-binding-operator.svc",
+                "service-binding-operator-webhook-service.service-binding-operator.svc.cluster.local",
+            ],
+            "secretName": "service-binding-operator-serving-cert",
+            "issuerRef": {
+                "kind": "ClusterIssuer",
+                "name": "selfsigned-issuer",
+            },
+        },
+    }
+
 def create_service_account():
     service_account = {
-    "apiVersion" : 'v1',
-    "kind" : 'ServiceAccount',
-    "metadata" : {
-        "name" : 'service-binding-operator',
-        "namespace" : 'service-binding-operator'
-    },
-}
+        "apiVersion" : 'v1',
+        "kind" : 'ServiceAccount',
+        "metadata" : {
+            "name" : 'service-binding-operator',
+            "namespace" : 'service-binding-operator'
+        },
+    }
     return service_account
 
+kind2prefix = {
```

Comment:
can we come up with some better name?

Reply:
How would you call a dictionary or map that maps a Kind to a Prefix? I'm not sure `kindToPrefixMap` or `kindToPrefixDict` would be much more descriptive.

Reply:
IMHO, you do not need this map at all, the most of the content can be simply copied from CSV, here is what is to be found there:

```yaml
webhookdefinitions:
  - admissionReviewVersions:
    - v1beta1
    containerPort: 443
    deploymentName: service-binding-operator
    failurePolicy: Fail
    generateName: vservicebinding.kb.io
    rules:
    - apiGroups:
      - binding.operators.coreos.com
      apiVersions:
      - v1alpha1
      operations:
      - CREATE
      - UPDATE
      resources:
      - servicebindings
    sideEffects: None
    targetPort: 9443
    type: ValidatingAdmissionWebhook
    webhookPath: /validate-binding-operators-coreos-com-v1alpha1-servicebinding
```

We only need to:

* rename `generateName` to `name`
* create `clientConfig` and use `webhookPath` as `path`

`type` should be used for deciding what kind of configuration will be made.